RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h

```diff
@@ -22,29 +23,41 @@ class TauIdMVAAuxiliaries {
     /// return chi2 of the leading track ==> deprecated? <==
     float tau_leadTrackChi2(const reco::PFTau& tau) const {
       float LeadingTracknormalizedChi2 = 0;
-      const reco::PFCandidatePtr& leadingPFCharged = tau.leadPFChargedHadrCand() ;
+      const reco::CandidatePtr& leadingPFCharged = tau.leadPFChargedHadrCand();
       if (leadingPFCharged.isNonnull()) {
-        reco::TrackRef tref = leadingPFCharged -> trackRef();
-        if (tref.isNonnull()) {
-          LeadingTracknormalizedChi2 = (float)(tref -> normalizedChi2());
+      	const reco::PFCandidate* pfcand = dynamic_cast<const reco::PFCandidate*>(leadingPFCharged.get());
+      	if (pfcand != nullptr) {
+          reco::TrackRef tref = pfcand->trackRef();
+          if (tref.isNonnull()) {
+            LeadingTracknormalizedChi2 = (float)(tref -> normalizedChi2());
+          }
+        }
+        else {
+          const pat::PackedCandidate* patcand = dynamic_cast<const pat::PackedCandidate*>(leadingPFCharged.get());
+          if (patcand != nullptr && patcand->hasTrackDetails()) {
```

Comment:
To be fully parallel with the current reconstruction, the leadChargedHadrCand() method will actually not return an actual candidate if the tau is only made from tracks (though this would in principle be possible for tau reconstruction on top of MiniAOD since lost tracks are of type candidate), so this check makes sense here.